src/js/game/LevelMVC/LevelModel.js

```diff
@@ -583,21 +592,37 @@ module.exports = class LevelModel {
     return entity.canPlaceBlock(blockAtPosition);
   }
 
+  canPlaceBlockDirection(blockType = "", entity , direction) {
+    if (entity.isOnBlock) {
+      return false;
+    }
+    let plane = this.getPlaneToPlaceOn(this.getMoveDirectionPosition(entity, direction), entity);
+    if (plane === this.groundPlane) {
+      if (blockType === "redstoneWire" || blockType.substring(0,5) === "rails" && this.groundPlane.getBlockAt(this.getMoveDirectionPosition(entity, direction))) {
```

Comment:
this logic _definitely_ needs to live somewhere else

Reply:
Moved it into a static method in LevelBlock.

This cool? 👍